includes/classes/db/mysql/query_factory.php

```diff
@@ -72,8 +72,15 @@ public function connect(string $db_host, string $db_user, string $db_password, s
         mysqli_report(MYSQLI_REPORT_OFF);
 
         $connectionRetry = 10;
-        while (!isset($this->link) || ($this->link == false && $connectionRetry != 0)) {
+        while (!isset($this->link) || ($this->link == false && $connectionRetry > 0)) {
             $this->link = mysqli_connect($db_host, $db_user, $db_password, $db_name, (defined('DB_PORT') ? DB_PORT : null), (defined('DB_SOCKET') ? DB_SOCKET : null));
+
+            // mariadb connection down or incorrect
+            if (mysqli_connect_errno() == 2002) {
+                if ($connectionRetry > 1) $connectionRetry = 1; // if service is down, try only one more time
```

Comment:
PSR, no single-line conditionals.

Reply:
Sigh. I really despise using 3-lines for a 1-line thing that's actually clearer to read when written as 1-line.
But, done.